fstest/fstests/fstests.go

```diff
@@ -176,13 +182,17 @@ func PutTestContentsMetadata(ctx context.Context, t *testing.T, f fs.Fs, file *f
 			}
 			obji.WithMetadata(metadata).WithMimeType(mimeType)
 		}
-		obj, err = f.Put(ctx, in, obji)
+		obj, err = f.Put(ctx, in, obji, options...)
 		return err
 	})
 	file.Hashes = uploadHash.Sums()
 	if check {
+		// Default mtime header for everything except b2
```

Comment:
I'm not sure I agree with this part...

In rclone's metadata scheme the `mtime` should **always** be the name of the key in the metadata and it is the job of the backend to translate it.

So I don't think this should have mtimeHeaderName passed into it.

Reply:
The issue here is that `b2` is the sole outlier in using `src_last_modified_millis` rather than `mtime`, behavior that is part of the documented interface for `b2`: https://rclone.org/b2/#modification-times.

The logic here seemed like a better option for now than reworking `b2` to use `mtime`. Reworking `b2` to use `mtime` as well as/instead of `src_last_modified_millis` seemed out of scope for a change focused on fixing upload headers for large files, and your earlier advice was to merge the `PutTestContentsOptions` function I broke out for `b2` back into `PutTestContentsMetadata`, so this is where I landed. Happy to move forward in whichever way you suggest!

Reply:
Hi @ncw - I should maybe have mentioned you with that implicit question, and made it more explicit: is it better, in the context of this PR, to:
* Change `b2` so that it supports `mtime` as well as `src_last_modified_millis`.
* Have `b2` indicate to `PutTestContentsMetadata` that it uses `src_last_modified_millis` rather than `mtime` (where the code is now).
* Have `b2` use a different function than `PutTestContentsMetadata` (where the code was).
* Some other solution I haven't thought of 😄

Reply:
He he - thank you for the list of options :-)

We don't want `src_last_modified_millis` to ever escape into the rclone metadata output. Your changes to `PutTestContentsMetadata` are a bit untidy (mixing different layers of the code) but don't cause this escape since you haven't implemented metadata yet.

For a full metadata implementation we'll need it to appear as `mtime` to the outside world but that must be saved as a parameter called `src_last_modified_millis` as that is part of the b2 protocol.

> 1. Change `b2` so that it supports `mtime` as well as `src_last_modified_millis`.

This won't be wasted work as it will be needed for metadata support.

> 2. Have `b2` indicate to `PutTestContentsMetadata` that it uses `src_last_modified_millis` rather than `mtime` (where the code is now).

I don't like the b2 specific code in a general routine. I've tried to keep backend specific stuff out of `fstest/fstests/fstests.go`  and only use the `Features()` flags.

> 3. Have `b2` use a different function than `PutTestContentsMetadata` (where the code was).

Will need to be undone later when metadata is supported!

> * Some other solution I haven't thought of 😄

:-)

My feeling is probably 1. if it isn't too hard, but 3. would be fine also.

It depends on what your appetite for doing the full metadata is for the b2 backend?